### PR TITLE
[00:35:20] Runtime in hostile.dm, line 352: Cannot read null.y

### PR DESCRIPTION
**Pull request by ExcessiveUseOfCobblestone**
Does it exist?

### CODE REVIEW

**Comment by coiax** [comment]
Do any of us?

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 1]
```diff
@@ -336,7 +336,7 @@
 
 
 /mob/living/simple_animal/hostile/proc/Shoot(atom/targeted_atom)
-	if(targeted_atom == targets_from.loc || targeted_atom == targets_from)
+	if(targeted_atom == targets_from.loc || targeted_atom == targets_from || targeted_atom == null)
```
Comment (GunHog) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: Use `!targeted_atom` as the first check. No point in trying to compare something that does not exist with other things before checking if that thing even exists!

Comment (Cyberboss) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: `QDELETED(targeted_atom)`

File: code/modules/mob/living/simple_animal/hostile/hostile.dm [hunk 2]
```diff
@@ -336,7 +336,7 @@
 
 
 /mob/living/simple_animal/hostile/proc/Shoot(atom/targeted_atom)
-	if(targeted_atom == targets_from.loc || targeted_atom == targets_from)
+	if( !QDELETED(targeted_atom) || targeted_atom == targets_from.loc || targeted_atom == targets_from )
```
Comment (lzimann) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: this actually needs to be just `QDELETED(...)` without the negation.

Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: Frick Cybernerd

Comment (ExcessiveUseOfCobblestone) [inline code/modules/mob/living/simple_animal/hostile/hostile.dm]: I'll fix it tomorrow, I'm in bed rn ;-)

**Pull request merged**